### PR TITLE
JS architecture update in NPM

### PR DESCRIPTION
**Pull request by ishachopra0401**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by MoumitaM — APPROVED** [review]
Don’t publish it yet. After our NPM package goes into production then only it should be published.

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/quick-start-guide.mdx [hunk 1]
```diff
@@ -111,12 +111,7 @@ To integrate and initialize the JavaScript SDK, you need the source write key an
 
 ### Alternative installation using NPM
 
-<div class="infoBlock">
-
-The NPM version of the JavaScript SDK is still based on the older architecture (v1). We are working on migrating it to the newer architecture.
-</div>
-
-Although we recommend using the snippets mentioned above to use the JavaScript SDK with your website, you can also use this [**NPM module**](https://www.npmjs.com/package/rudder-sdk-js) to package RudderStack directly into your project.
+It is recommended to use the snippets mentioned above to use the JavaScript SDK with your website. However, you can also use this [**NPM module**](https://www.npmjs.com/package/rudder-sdk-js) with the new JavaScript SDK v1.1 architecture to package RudderStack directly into your project.
```
Comment (ameypv-rudder) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/quick-start-guide.mdx:114]:
```suggestion
While it is recommended to use the above-mentioned snippets to use the JavaScript SDK with your website, you can also use the [**NPM module**](https://www.npmjs.com/package/rudder-sdk-js) with the new SDK(v1.1) architecture to package RudderStack directly into your project.
```

Comment (ameypv-rudder) [reply stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/quick-start-guide.mdx]: Should we also reference this somewhere: "Integrations will be loaded dynamically." cc @MoumitaM something along the lines of:

"...to page RudderStack directly into your project and load the integrations dynamically."

If the sentence is too long, maybe we can put the recommendation as a note and start the section with 

"You can also use the NPM module with the new SDK architecture to package RudderStack directly into your project and load the integrations dynamically."

**Comment by ameypv-rudder** [comment]
Should we also reference this somewhere: "Integrations will be loaded dynamically." cc @MoumitaM something along the lines of:

"...to page RudderStack directly into your project and load the integrations dynamically."

If the sentence is too long, maybe we can put the recommendation as a note and start the section with

"You can also use the NPM module with the new SDK architecture to package RudderStack directly into your project and load the integrations dynamically."

**Comment by ishachopra0401** [comment]
> Should we also reference this somewhere: "Integrations will be loaded dynamically." cc @MoumitaM something along the lines of:
> 
> "...to page RudderStack directly into your project and load the integrations dynamically."
> 
> If the sentence is too long, maybe we can put the recommendation as a note and start the section with
> 
> "You can also use the NPM module with the new SDK architecture to package RudderStack directly into your project and load the integrations dynamically." cc @ishachopra0401 thoughts?

IMO if this is the default behavior, meaning the 'Integrations should load dynamically', we might skip it. Otherwise, if it's a feature, we should add it. AFAIU when we were at the older version, it didn't load dynamically (different behavior than expected). Now it should be the expected one, correct me if I am wrong.

**Comment by ameypv-rudder** [comment]
> 

makes sense. @MoumitaM wdyt?

**Comment by MoumitaM** [comment]
I think we should mention it. It is a breaking change. 

`"You can also use the NPM module with the new SDK architecture to package RudderStack directly into your project where integrations will be loaded dynamically."`

Modified the sentence a bit. 

 @ameypv-rudder, Do you think we should add some info on the main JS page? How would they know that we have published a new version with a breaking change?
CC @ishachopra0401

**Comment by ishachopra0401** [comment]
Incorporated the suggested changes.

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/index.mdx [hunk 2]
```diff
@@ -32,6 +32,11 @@ The JavaScript SDK now supports integration with the OneTrust consent manager an
 
 To quickly set up and start using the RudderStack JavaScript SDK, go through our [**quick start guide**](https://docs.rudderstack.com/stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/quick-start-guide).
 
+<div class="infoBlock">
+
+You can also use the NPM module with the new SDK (v1.1) architecture to package RudderStack directly into your project. For more information, refer to <a href="https://www.rudderstack.com/docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/quick-start-guide/#alternative-installation-using-npm">Install using NPM</a>.
```
Comment (ameypv-rudder) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/index.mdx:37]:
```suggestion
You can also use the NPM module with the new SDK(v1.1) architecture to package RudderStack directly into your project. For more information, refer to the <a href="https://www.rudderstack.com/docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/quick-start-guide/#alternative-installation-using-npm">Installation using NPM</a> section.
```

**Review by ameypv-rudder — APPROVED** [review]

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/index.mdx [hunk 3]
```diff
@@ -32,6 +32,11 @@ The JavaScript SDK now supports integration with the OneTrust consent manager an
 
 To quickly set up and start using the RudderStack JavaScript SDK, go through our [**quick start guide**](https://docs.rudderstack.com/stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/quick-start-guide).
 
+<div class="infoBlock">
```
Comment (saikumarrs) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/index.mdx:35]: This is not required. Let's remove it.

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/quick-start-guide.mdx [hunk 4]
```diff
@@ -111,12 +111,7 @@ To integrate and initialize the JavaScript SDK, you need the source write key an
 
 ### Alternative installation using NPM
 
-<div class="infoBlock">
-
-The NPM version of the JavaScript SDK is still based on the older architecture (v1). We are working on migrating it to the newer architecture.
-</div>
-
-Although we recommend using the snippets mentioned above to use the JavaScript SDK with your website, you can also use this [**NPM module**](https://www.npmjs.com/package/rudder-sdk-js) to package RudderStack directly into your project.
+While it is recommended to use the above-mentioned snippets to use the JavaScript SDK with your website, you can also use the [**NPM module**](https://www.npmjs.com/package/rudder-sdk-js) with the new SDK(v1.1) architecture to package RudderStack directly into your project where the integrations will be loaded dynamically.
```
Comment (saikumarrs) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/quick-start-guide.mdx:114]: As we've already declared that the new architecture is the default.
```suggestion
While it is recommended to use the above-mentioned snippets to use the JavaScript SDK with your website, you can also use the [**NPM module**](https://www.npmjs.com/package/rudder-sdk-js) to package RudderStack directly into your project.
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/index.mdx (hunk 3 above)
Comment (ishachopra0401) [reply stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/index.mdx]: done

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/quick-start-guide.mdx (hunk 4 above)
Comment (ishachopra0401) [reply stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/quick-start-guide.mdx]: updated

**Review by saikumarrs — APPROVED** [review]

**Comment by ameypv-rudder** [comment]
yes the PR is tagged with the appropriate label already, @saikumarrs

**Comment by saikumarrs** [comment]
@ishachopra0401 @ameypv-rudder Let's deploy these changes.

**Pull request merged**